syntax = "proto3";

package product;

service ProductService {
  rpc GetProducts (Empty) returns (ProductList) {}
  rpc GetProduct (ProductId) returns (Product) {}
  rpc CreateProduct (Product) returns (Product) {}
  rpc UpdateProduct (Product) returns (Product) {}
  rpc DeleteProduct (ProductId) returns (Empty) {}
}

message Empty {}

message ProductId {
  string id = 1;
}

message Product {
  string id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
  string image_url = 5;
  bool available = 6;
}

message ProductList {
  repeated Product products = 1;
}
